src/Symfony/Bundle/SecurityBundle/Security.php

```diff
@@ -60,8 +62,20 @@ public function getUser(): ?UserInterface
      */
     public function isGranted(mixed $attributes, mixed $subject = null): bool
     {
-        return $this->container->get('security.authorization_checker')
-            ->isGranted($attributes, $subject);
+        return $this->getDecision($attributes, $subject)->isGranted();
+    }
+
+    /**
+     * Get the access decision against the current authentication token and optionally supplied subject.
+     */
+    public function getDecision(mixed $attribute, mixed $subject = null): AccessDecision
+    {
+        $checker = $this->container->get('security.authorization_checker');
+        if (method_exists($checker, 'getDecision')) {
```

Comment:
let's swap the returns to reduce the diff with future versions:
```suggestion
        if (!method_exists($checker, 'getDecision')) {
```